### ISSUE TITLE
client.zerotier.list() randomly crash

### ISSUE THREAD

**Issue by maxux** [zero-os/0-core#496]
Some (random) time, when executing `client.zerotier.list()`, the call just crash with this error:
```
  File "/home/gig/js9.1.2/lib/python3.5/site-packages/zeroos/core0/client/client.py", line 880, in list
    return self._client.json('corex.zerotier.list', {'container': self._container})
  File "/home/gig/js9.1.2/lib/python3.5/site-packages/zeroos/core0/client/client.py", line 762, in json
    result = self.sync(command, arguments, tags=tags, id=id)
  File "/home/gig/js9.1.2/lib/python3.5/site-packages/zeroos/core0/client/client.py", line 752, in sync
    raise RuntimeError('invalid response: %s' % result.state, result)
RuntimeError: ('invalid response: ERROR', STATE: 1002 ERROR
STDOUT:

STDERR:

DATA:
"(ERROR): [ zerotier-cli: missing port and zerotier-one.port not found in /var/cache/containers/container-1/zerotier\n]"
)
```

**Comment by muhamadazmy** [comment]
It means that the zt daemon of that container has crashed. I will look into it.

**Comment by muhamadazmy** [comment]
I was working on a fix that re-spawn zerotier of a container if it died unexpectedly. Accidentally I managed to reproduce this bug (which I couldn't do before) It only happens if this node was added to an orchestrator grid. What happens is the following:

- You start the node
- You create a container
- Ays picks up the node (after reboot or discovered via the node zerotier)
- Ays mounts the cache partition
- It mounts over the container backend under /var/cache/container which hides the zerotier home for that container and causes the list above to fail.

I already implemented a fix (not merged yet) to respawn zerotier if it died. But if the above situation is what trigger the issue then the respawn mechanism will not work as expected because the zerotier daemon itself never died, it's the zt home dir is what is hidden from the system hence it can't execute a join, list or leave commands.

Can u confirm the above scenario.

**Comment by FastGeert** [comment]
The node is not added to an orchestrator. The node is being used to run an orchestrator in a container.

One of the actions done by the installation script, mount you describe

**Comment by muhamadazmy** [comment]
Waiting for input from @maxux  to state the actual state of the container zerotier daemon and the mounts.

**Comment by maxux** [comment]
We just had the same issue. I checked the machine, nothing is mounted to `/var/cache/containers` (it's still the tmpfs), so no mounts are involved.
All I know is, during the autobootstrap script, it fails but when doing manually `client.zerotier.list()` couple of minutes later, the list seems correct. I guess the call is made too quickly after container creation ?

**Comment by maxux** [comment]
We just had the same issue. There is no mounts on `/var/cache/containers`, so it's the plain tmpfs, no override occured.

Here are manual detection:
```
~ # ls /var/cache/containers/container-1/zerotier/*
/var/cache/containers/container-1/zerotier/authtoken.secret
/var/cache/containers/container-1/zerotier/identity.public
/var/cache/containers/container-1/zerotier/identity.secret
/var/cache/containers/container-1/zerotier/planet

/var/cache/containers/container-1/zerotier/iddb.d:
8841408a2e  9d219039f3

~ # zerotier-cli -D/var/cache/containers/container-1/zerotier/ listnetworks
zerotier-cli: missing port and zerotier-one.port not found in /var/cache/containers/container-1/zerotier/
```

**Comment by muhamadazmy** [comment]
If this is just the zerotier crashes, then my PR https://github.com/zero-os/0-core/pull/504 should fix it. 

- If this crash happens on container creation you should get an error from the create API.
- If the error happened later during the life of the container, then it will respawn and your networks will be rejoined automatically.

**Comment by muhamadazmy** [comment]
Fixed by #504
